### PR TITLE
compare original snapshot with itself in 1st execution

### PR DESCRIPTION
**Pull request by KoditkarVedant**
Summary of the changes (Less than 80 chars)

The 'assert' or 'typeof' match options are only applied if a snapshot already exists. As per discussion, we will compare original snapshot with itself during first run.

Note: There are no changes to SNAPSHOOTER_STRICT_MODE "on" - it will continue to throw an exception if the snapshot is not present.

Addresses #80

### CODE REVIEW

File: src/Snapshooter/SnapshotAssert.cs [hunk 1]
```diff
@@ -88,13 +88,17 @@ public class SnapshotAssert : ISnapshotAssert
                         "rerun your tests.");
                 }
 
-                _snapshotFileHandler
-                    .SaveNewSnapshot(snapshotFullName, actualSnapshotSerialized);
-                return;
+                _snapshotFileHandler.SaveNewSnapshot(snapshotFullName, actualSnapshotSerialized);
+
+                savedSnapshotSerialized = _snapshotFileHandler.ReadSnapshot(snapshotFullName);
```
Comment (KoditkarVedant) [inline src/Snapshooter/SnapshotAssert.cs:93]: I'm not sure how can I add test cases for this change. For the sake, I can use Moq to verify if `_snapshotFileHandler.ReadSnapshot` was called twice and `_snapshotComparer.CompareSnapshots` called once.

If anyone can guide me I can work on that.

**Comment by KoditkarVedant** [comment]
@nscheibe tagging you in case if we missed this PR.

**Review by nscheibe — APPROVED** [review]
Looks good. Everything under test.

**Review by nscheibe — APPROVED** [review]
Done

**Comment by michaelstaib** [comment]
The build failed, there seems to be something wrong with the sonar token.  I can look at this tomorrow.

**Comment by nscheibe** [comment]
Try to restart pr build

**Review by nscheibe — APPROVED** [review]
Everything seems to be fine...

**Comment by nscheibe** [comment]
I merged it, also if our PR always failed... there is something wrong with the PR Build.

**Comment by nscheibe** [comment]
Thanks a lot @KoditkarVedant for your help/contribution. Very appreciated!

**Pull request merged**